### PR TITLE
sjb/children.sh: lookup parents only once

### PR DESCRIPTION
**Pull request by bbguimaraes**
Checked with:

```sh
$ for x in sjb/config/test_cases/*.yml sjb/config/common/test_cases/*.yml; do
> diff <(sjb/children.sh "$x") <(sjb/children.sh.old "$x")
> done
$ TIMEFORMAT='%R %U %S'
$ time sjb/children.sh.old sjb/config/common/test_cases/origin_minimal.yml
125.569 60.926 71.239
$ time sjb/children.sh sjb/config/common/test_cases/origin_minimal.yml
1.204 1.000 0.214
```

@stevekuznetsov: it had to be done.

/me awaits the impending bash code review.

### CODE REVIEW

File: sjb/children.sh [hunk 1]
```diff
@@ -4,21 +4,24 @@ set -o errexit
...
 
-function children() {
-	local config=$1
-
-	for child_config in sjb/config/test_cases/*.yml sjb/config/common/test_cases/*.yml; do
-		if ! grep -q 'parent' "${child_config}"; then
-			continue
-		fi
+CONFIGS='sjb/config/test_cases/*.yml sjb/config/common/test_cases/*.yml'
+declare -A PARENTS
```
Comment (stevekuznetsov) [inline sjb/children.sh]: This is not portable and won't work on latest MacOS AFAIR

File: sjb/children.sh [hunk 2]
```diff
@@ -4,21 +4,24 @@ set -o errexit
...
+            echo "${child_config}"
+            children "${child_config}"
+        fi
+    done
 }
 
-children "$1"
\ No newline at end of file
+for config in $CONFIGS; do
+    if parent=$(grep -Po "(?<=parent: ')[^']+(?=')" "${config}"); then
```
Comment (stevekuznetsov) [inline sjb/children.sh]: We could use Python and `yaml.load()` :)

**Comment by stevekuznetsov** [comment]
I am generally fine with this but worry someone using a mac will break on it

File: sjb/children.sh (hunk 1 above)
Comment (bbguimaraes) [reply sjb/children.sh]: Who uses mac anyw…

File: sjb/children.sh (hunk 2 above)
Comment (bbguimaraes) [reply sjb/children.sh]: See below.

**Comment by stevekuznetsov** [comment]
Cool, let's use the regex then since it's not built-in. Thanks for looking at this

**Comment by bbguimaraes** [comment]
@stevekuznetsov: updated.

You can thank my performance OCD =)

**Pull request merged**